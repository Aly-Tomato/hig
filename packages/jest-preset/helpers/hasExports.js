module.exports = function hasExports(index, cases) {
  cases.forEach(({ name, value }) => {
    it(`exports ${name}`, () => {
      expect(index).toHaveProperty(name, value);
    });
  });
};
