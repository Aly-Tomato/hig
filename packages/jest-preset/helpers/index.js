/* eslint-disable global-require */
module.exports = {
  hasExports: require("./hasExports"),
  lastCallOfMock: require("./lastCallOfMock"),
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
