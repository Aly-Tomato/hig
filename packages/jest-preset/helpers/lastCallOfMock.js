module.exports = function lastCallOfMock(mockFn) {
  return mockFn.mock.calls[mockFn.mock.calls.length - 1];
};
