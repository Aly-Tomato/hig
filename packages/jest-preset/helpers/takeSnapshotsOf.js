const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const presenter = React.createElement(Component, props);
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
