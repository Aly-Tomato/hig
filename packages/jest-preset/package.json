{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "dependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.2",
    "enzyme": "^2.9.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.1",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.2"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
