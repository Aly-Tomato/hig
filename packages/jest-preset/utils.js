require("@hig/utils").generateId = jest.fn(key => `${key}-1`);
