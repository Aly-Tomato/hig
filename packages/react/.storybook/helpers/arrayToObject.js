export default function arrayToObject(arr) {
  var rv = {};
  for (var i = 0; i < arr.length; ++i) rv[arr[i]] = arr[i];
  return rv;
}
