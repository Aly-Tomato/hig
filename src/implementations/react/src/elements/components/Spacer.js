import React from "react";
import SpacerAdapter from "../../adapters/SpacerAdapter";

class Spacer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  setSlotEl = el => {
    this.setState({ setSlotEl: el });
  };

  render() {
    return (
      <SpacerAdapter {...this.props}>
        <div ref={this.setSlotEl}>{this.props.children}</div>
      </SpacerAdapter>
    );
  }
}

export default Spacer;
