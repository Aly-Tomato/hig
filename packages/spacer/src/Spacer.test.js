import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import Spacer from "./Spacer";

describe("spacer/Spacer", () => {
  takeSnapshotsOf(Spacer, [
    {
      description: "renders with no props",
      props: {}
    },
    {
      description: "renders with spacing",
      props: { spacing: "m" }
    },
    {
      description: "renders with size",
      props: { size: "40px" }
    },
    {
      description: "renders with size and spacing",
      props: { size: "40px", spacing: "m" }
    }
  ]);
});
