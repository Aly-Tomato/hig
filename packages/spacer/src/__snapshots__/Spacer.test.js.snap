// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacer/Spacer  1`] = `
.emotion-0 {
  width: 16px;
  height: 16px;
}

<div
  className="emotion-0 hig__spacer-v1"
/>
`;

exports[`spacer/Spacer  2`] = `
.emotion-0 {
  width: 16px;
  height: 16px;
}

<div
  className="emotion-0 hig__spacer-v1"
/>
`;

exports[`spacer/Spacer  3`] = `
.emotion-0 {
  width: 40px;
  height: 40px;
}

<div
  className="emotion-0 hig__spacer-v1"
/>
`;

exports[`spacer/Spacer  4`] = `
.emotion-0 {
  width: 40px;
  height: 40px;
}

<div
  className="emotion-0 hig__spacer-v1"
/>
`;
